src/resolve-system-dependencies/resolve-system-dependencies.cc

```diff
@@ -0,0 +1,204 @@
+#include "derivations.hh"
+#include "globals.hh"
+#include "shared.hh"
+#include "store-api.hh"
+#include <sys/utsname.h>
+#include <algorithm>
+#include <iostream>
+#include <fstream>
+#include <mach-o/loader.h>
+#include <mach-o/swap.h>
+
+using namespace nix;
+
+static auto cacheDir = Path{};
+
+Path resolveCacheFile(Path lib) {
+    std::replace(lib.begin(), lib.end(), '/', '%');
+    return cacheDir + "/" + lib;
+}
+
+std::set<string> readCacheFile(const Path & file) {
+    return tokenizeString<set<string>>(readFile(file), "\n");
+}
+
+void writeCacheFile(const Path & file, std::set<string> & deps) {
+    std::ofstream fp;
+    fp.open(file);
+    for (auto & d : deps) {
+        fp << d << "\n";
+    }
+    fp.close();
+}
+
+std::string findDylibName(FILE *obj_file, struct load_command cmd) {
+    fpos_t pos;
+    fgetpos(obj_file, &pos);
+    struct dylib_command dylc;
+    dylc.cmd = cmd.cmd;
+    dylc.cmdsize = cmd.cmdsize;
+    fread(&dylc.dylib, sizeof(struct dylib), 1, obj_file);
+
+    char *dylib_name = (char*)calloc(cmd.cmdsize, sizeof(char));
+    fseek(obj_file,
+            // offset is calculated from the beginning of the load command, which is two
+            // uint32_t's backwards
+            dylc.dylib.name.offset - (sizeof(uint32_t) * 2) + pos,
+            SEEK_SET);
+    fread(dylib_name, sizeof(char), cmd.cmdsize, obj_file);
+    fseek(obj_file, pos, SEEK_SET);
```

Comment:
This lacks error checking. I would use C++ `<iostream>`, or read the whole file into memory using `readFile()` for simplicity.